packages/story-editor/src/components/library/panes/text/textSets/textSetsPane.js

```diff
@@ -172,22 +174,25 @@ function TextSetsPane({ paneRef }) {
       localStore.setItemByKey(`${LOCAL_STORAGE_PREFIX.TEXT_SET_SETTINGS}`, {
         selectedCategory,
       });
+      trackChange.current = true;
     },
     [speak]
   );
 
-  const onChangeShowInUse = useCallback(
-    () => requestAnimationFrame(() => setShowInUse((prevVal) => !prevVal)),
-    [setShowInUse]
-  );
+  const onChangeShowInUse = useCallback(() => {
+    requestAnimationFrame(() => setShowInUse((prevVal) => !prevVal));
+    trackChange.current = true;
+  }, [setShowInUse]);
 
   useEffect(() => {
-    trackEvent('search', {
-      search_type: 'textsets',
-      search_term: '',
-      search_category: selectedCat,
-      search_filter: showInUse ? 'show_in_use' : undefined,
-    });
+    if (selectedCat || showInUse || trackChange.current) {
+      trackEvent('search', {
+        search_type: 'textsets',
+        search_term: '',
+        search_category: selectedCat,
+        search_filter: showInUse ? 'show_in_use' : undefined,
+      });
+    }
```

Comment:
@swissspidy The way I have made this work is that when you open the `textSet` tab, if you had previously changed the textSet filtering this event will still fire (we are storing those filtering setting in localStorage). Is that the desired behavior? Or do we only want to track when the user actively changes the filtering setting?

Also, not sure if we want to rename this event sense we are no longer using the search input on textSet, only filtering with text that match fonts from story and via categories. ie. `cover` `steps` etc.

Reply:
> if you had previously changed the textSet filtering this event will still fire (we are storing those filtering setting in localStorage). Is that the desired behavior? Or do we only want to track when the user actively changes the filtering setting?

Ideally only on user interaction, but if that's too big of a change it should be fine to keep I think.

> Also, not sure if we want to rename this event sense we are no longer using the search input on textSet

We're good; `search` has a special meaning in analytics that we wanna keep.